lib/plivo/resources/verify_session.rb

```diff
@@ -0,0 +1,136 @@
+                    valid_param?(:channel, channel, [String, Symbol], false)
+                    valid_param?(:url, url, [String], false)
+                    valid_param?(:method, method, String, false, %w[POST GET])
+
+                    params = {
+                        app_uuid: app_uuid,
+                        recipient: recipient,
+                        channel: channel,
+                        url: url,
+                        method: method
+                    }
+                    
+                end    
+                perform_create(params)    
+            end   
+            
+            def list(options = nil)
+                return perform_list if options.nil?
+                valid_param?(:options, options, Hash, true)
+                params = {}
+                params_expected = %i[
+                subaccount status session_time__gt session_time__gte
+                session_time__lt session_time__lte session_time country alias app_uuid recipient
+                ]
+
+                params_expected.each do |param|
+                    if options.key?(param) &&
+                       valid_param?(param, options[param], [String, Symbol], true)
+                      params[param] = options[param]
+                    end
+                end
+          
+                %i[offset limit].each do |param|
+                    if options.key?(param) &&
+                        valid_param?(param, options[param], [Integer, Integer], true)
+                        params[param] = options[param]
+                    end
+                end
+                
+                if options.key?(:limit) &&
+                    (options[:limit] > 20 || options[:limit] <= 0)
+                   raise_invalid_request('The maximum number of results that can be '\
+                   "fetched is 20. limit can't be more than 20 or less than 1")
+                end
+
+                raise_invalid_request("Offset can't be negative") if options.key?(:offset) && options[:offset] < 0
+
+                perform_list_without_object(params)
+            end
+            
+            def validate(session_uuid = nil, otp = nil)
+                value = session_uuid
+                id = ""
+                if(value.is_a?(Hash))
+                    valid_param?(:session_uuid, value[:session_uuid], [String, Symbol], true)
+                    valid_param?(:otp, value[:otp], [String], true)  
+                    id = value[:session_uuid]
+                    params = {
+                        otp: value[:otp]
+                    }
+                else
+                    valid_param?(:session_uuid, session_uuid, [String, Symbol], true)
+                    valid_param?(:otp, otp, [String], true)  
+                    id = session_uuid
+                    params = {
+                        otp: otp
+                    }
+                end  
+                perform_create_with_id(id,params)      
```

Comment:
can we use perform_action? rather than adding new function

Reply:
yes